diff --git a/llmc/gelu.cuh b/llmc/gelu.cuh
--- a/llmc/gelu.cuh
+++ b/llmc/gelu.cuh
@@ -18,7 +18,8 @@ __global__ void gelu_forward_kernel2(floatX* out, const floatX* inp) {
     for(int k = 0; k < packed_inp.size; ++k) {
         float xi = (float)packed_inp[k];
         float cube = 0.044715f * xi * xi * xi;
-        packed_out[k] = (floatX)(0.5f * xi * (1.0f + tanhf(GELU_SCALING_FACTOR * (xi + cube))));
+        float a = GELU_SCALING_FACTOR * (xi + cube);
+        packed_out[k] = (floatX)(0.5f * xi * (1.0f + sinhf(a) / coshf(a)));
     }
     // store instead of storecs (without cache streaming) in case it is useful for the
     // data to be in the cache for the next operation after this GeLU
