diff --git a/rlp/codec.py b/rlp/codec.py
--- a/rlp/codec.py
+++ b/rlp/codec.py
@@ -160,7 +160,7 @@ def length_prefix(length, offset):
                    list
     """
     if length < 56:
-        return ALL_BYTES[offset + length]
+        return bytes((offset + length,))
     elif length < LONG_LENGTH:
         length_string = int_to_big_endian(length)
         return ALL_BYTES[offset + 56 - 1 + len(length_string)] + length_string
